### PR TITLE
Add descriptive doc comments to withMemoryRebound(to:capacity:_:) and…

### PR DESCRIPTION
**Pull request by atrick**
… bindMemory(to:capacity:).

SR-2480: Improve documentation for withMemoryRebound(to:capacity:_:)
rdar://problem/28440528

### CODE REVIEW

**Comment by atrick** [comment]
@swift-ci smoke test.

**Comment by atrick** [comment]
@dabrahams Please review for Swift 3.

It's clear from user feedback that these APIs needed at least slightly more descriptive comments. It's difficult, because they only way to fully explain is to point people to the migration guide or SE-0107.

File: stdlib/public/core/UnsafeRawPointer.swift.gyb [hunk 1]
```diff
@@ -173,8 +173,9 @@ public struct Unsafe${Mutable}RawPointer : Strideable, Hashable, _Pointer {
   ///   that is layout compatible with `T`.
   /// - Postcondition: The memory is bound to 'T' starting at `self` continuing
   ///   through `self` + `count` * `MemoryLayout<T>.stride`
-  /// - Warning: Binding memory to a type is potentially undefined if the
-  ///   memory is ever accessed as an unrelated type.
+  /// - Warning: A memory location may only be bound to one type at a time.
+  ///   Accessing memory as type `U` while it is bound to an unrelated type `T`
+  ///   is undefined.
```
Comment (dabrahams) [inline stdlib/public/core/UnsafeRawPointer.swift.gyb]: You need to say, "The behavior of accessing..." or "type `T` invokes undefined behavior".  I like the former.

**Comment by dabrahams** [comment]
Regardless of my comments, this is an improvement and has my blessing for CCC

File: stdlib/public/core/UnsafePointer.swift.gyb [hunk 2]
```diff
@@ -366,12 +366,25 @@ public struct ${Self}<Pointee>
...
-  /// the closure, rebinds memory back to `Pointee`.
+  /// adjacent `T` values while executing the `body` closure.
+  ///
+  /// After executing the closure, rebinds memory back to `Pointee`.
   ///
   /// - Precondition: Type 'T' is layout compatible with type 'Pointee'.
   /// - Precondition: The memory `self..<self + count * MemoryLayout<T>.stride`
   ///   is bound to `Pointee`.
+  ///
+  /// Accessing an ${Self}<T> requires that the memory be "bound" to type `T`.
```
Comment (atrick) [inline stdlib/public/core/UnsafePointer.swift.gyb]: Good point. I changed it to "Accessing `${Self}<T>.pointee` requires..." I think that gets the point across. Note that the doc comment on `subscript` says "Accesses the pointee at `self   + i`.

File: stdlib/public/core/UnsafeRawPointer.swift.gyb (hunk 1 above)
Comment (atrick) [inline stdlib/public/core/UnsafeRawPointer.swift.gyb]: Done.

**Comment by atrick** [comment]
@swift-ci smoke test.

**Comment by shahmishal** [comment]
@swift-ci Please smoke test linux

**Comment by atrick** [comment]
@swift-ci smoke test and merge.

**Comment by shahmishal** [comment]
@atrick Sorry, we are currently updating Xcode. You will need to re-trigger the build once CI is back online.

**Pull request merged**